### PR TITLE
create a pointer from extracted elements back to the bs4 tag

### PR DESCRIPTION
**Pull request by avyfain**
Add a reference back to the BeautifulSoup tag to let users easily find the location of a given tag in its surrounding HTML.

Also adds the default `.gitignore` from https://github.com/github/gitignore/blob/main/Python.gitignore

### CODE REVIEW

**Comment by avyfain** [comment]
Hey there! I made a small update here to scratch my own itch - first time contributing to the project, so let me know if there's anything else I should be keeping an eye out for.

Both tests and mypy run fine locally on my end, although I think the tests could be improved (ie, shouldn't be testing `is not None` but something else).

**Comment by drkane** [comment]
This is a great addition - thanks. It looks good to me but I'll take a proper look and then merge.

File: tests/test_parse.py
Comment (drkane) [inline tests/test_parse.py:1]: Could we change

```python
assert x.nonnumeric[0].soup_tag is not None
```

to
```python
assert isinstance(x.nonnumeric[0].soup_tag, Tag)
```

to make the tests more meaningful?

Comment (avyfain) [reply tests/test_parse.py:1]: Good call!

**Comment by avyfain** [comment]
Super cool to see this merged. Thanks, David!

**Pull request merged**